internal/testrunner/runners/system/runner.go

```diff
@@ -478,6 +478,11 @@ func (r *runner) runTest(config *testConfig, ctxt servicedeployer.ServiceContext
 		}
 	}
 
+	// Check Hit Count within docs, if 0 then it has not been specified
```

Comment:
Up on line 446 there is code to wait for documents to be received.

```go
docs, err = r.getDocs(dataStream)
return len(docs) > 0, err
```

If I am using `assert.hit_count` then my expectation is that the test will wait for at least the expected number of hits to be received. Imagine a test where the documents are read in slowly (say it takes 30 seconds to pull down the expected documents and they are posted in a few separate _bulks). Currently this will stop waiting as soon as any documents are received. Then when the count is checked here it will be wrong because the it didn't wait long enough. I think the check on L446 needs to be amended to wait.

Reply:
Double checking logic here: I am accustomed to assert in the sense of "flag an error if not condition satisfied" but what you actually want is "keep doing action until condition satisfied" correct?

Reply:
It should do both. If it the test runner doesn't wait until the condition can be satisfied then it will be brittle.

Reply:
I'm really not getting this. Assertions would normally run after a test, not during. If an assertion evaluates to not true, that test has ended and is flagged as a fail. However, I think what you want is to keep running the "assertion" as part of the mechanics of test setup until enough docs/hits have arrived & only to flag a fail if, after trying "n" times, the "assertion" fails. If the loop repeats until the assertion is true then the assertion will never fail. @andrewkroh this is why I am suggesting to separate the loop exit condition from the assertion, even if they share the same target quantity - does this make sense?

Reply:
When employing this assertion, I think the wait condition for the test running needs to be `len(docs) >= config.Assert.HitCount`.

Reply:
Hi Andrew, I'm not asking how to compare a greater than or equal value, my question is whether it's OK to continue a test after an assert has failed, or alternatively do the `>=` comparison separately to the assert. I appreciate they will have the same target value in some use cases, it's more a point of the nature of an assertion

Reply:
I'm not asking to have the call to `assertHitCount` moved. It's good where it's at.

I want this closure to wait for at least the expected number of hits to be received when the `assert.hit_count` feature is in use.
https://github.com/elastic/elastic-package/blob/74bc7061097d3cab12506b78674e1bcd5ed39d50/internal/testrunner/runners/system/runner.go#L446-L453